spyder/utils/programs.py

```diff
@@ -82,42 +82,28 @@ def is_program_installed(basename):
     """
     home = get_home_dir()
     req_paths = []
-    if sys.platform == 'darwin':
-        if basename.endswith('.app') and osp.exists(basename):
-            return basename
-
-        pyenv = [
-            osp.join('/usr', 'local', 'bin'),
-            osp.join(home, '.pyenv', 'bin')
-        ]
-
-        # Prioritize Anaconda before Miniconda; local before global.
-        a = [osp.join(home, 'opt'), '/opt']
-        b = ['anaconda', 'miniconda', 'anaconda3', 'miniconda3']
-        conda = [osp.join(*p, 'condabin') for p in itertools.product(a, b)]
+    if (sys.platform == 'darwin' and basename.endswith('.app')
+            and osp.exists(basename)):
+        return basename
 
-        req_paths.extend(pyenv + conda)
-
-    elif sys.platform.startswith('linux'):
+    if os.name == 'posix':
         pyenv = [
+            osp.join(home, '.pyenv', 'bin'),
             osp.join('/usr', 'local', 'bin'),
-            osp.join(home, '.pyenv', 'bin')
         ]
 
         a = [home, '/opt']
-        b = ['anaconda', 'miniconda', 'anaconda3', 'miniconda3']
-        conda = [osp.join(*p, 'condabin') for p in itertools.product(a, b)]
-
-        req_paths.extend(pyenv + conda)
-
-    elif os.name == 'nt':
+        b = ['mambaforge', 'miniforge',
+             'miniconda3', 'anaconda3', 'miniconda', 'anaconda']
+    else:
         pyenv = [osp.join(home, '.pyenv', 'pyenv-win', 'bin')]
 
         a = [home, 'C:\\', osp.join('C:\\', 'ProgramData')]
```

Comment:
```suggestion
        a = [home, osp.join(home, 'AppData/Local'),
             'C:/', osp.join('C:/', 'ProgramData')]
```

At least Miniforge and Miniconda default to being installed in `~/AppData/Local` instead of `~` on Windows.

Reply:
Mambaforge is in `~\AppData\Local` local as well. Do you think Anaconda is too?
Maybe `home` should be removed?
```suggestion
        a = [osp.join(home, 'AppData/Local'), osp.join('C:/', 'ProgramData')]
```

Reply:
Or can other programs be found in `home` or `C:/`?

Reply:
Sorry, I meant Mambaforge and Miniforge. I've seen Miniconda, etc. in `~` quite commonly on Windows, so that should stay. And the drive root is a common location (I use it and have seen others do the same, for instance, since my username includes spaces which causes problems with some stuff, and IIRC some guides do too).